src/main/java/com/ledikom/service/BotService.java

```diff
@@ -66,6 +67,19 @@ public void processCouponsInMap() {
         CouponService.userCoupons.entrySet().removeIf(userCoupon -> userCoupon.getValue().getExpiryTimestamp() < System.currentTimeMillis() - 5000);
     }
 
+    @Scheduled(cron = "0 0 0 * * *")
+    public void deleteCouponsIfExpired() {
+        List<Coupon> coupons = couponService.getAllCoupons();
+
+        LocalDateTime currentDateTime = LocalDateTime.now();
+
+        for (Coupon coupon : coupons) {
+            if (coupon.getId() != 1 && currentDateTime.isAfter(coupon.getExpiryDateTime())) {
```

Comment:
helloCoupon is found not by id=1 (it can change, so will not work),
it is found by `hello-coupon.name` from config

Can you also please add uniqueness to name in Coupon entity, so the finding by name will be secured?